qiskit/tools/visualization/interactive/_iplot_blochsphere.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Bloch sphere visualization
+"""
+from string import Template
+import time
+import re
+import numpy as np
+from qiskit.tools.qi.pauli import pauli_singles
+try:
+    from IPython.core.display import display, HTML
```

Comment:
All of these import statements work in a normal IPython terminal env, and thus do not actually check for a notebook env.  Instead, try this:

```python
import sys
if ('ipykernel' in sys.modules) and ('spyder' not in sys.modules):
    from IPython.core.display import display, HTML
```

This covers everything except for the edge case of someone using the Qtconsole.  But I have yet to run into someone actually using that now that notebooks exist.